restapi/src/main/java/io/stargate/web/resources/AuthenticatedDB.java

```diff
@@ -106,4 +124,21 @@ public UserDefinedType getType(String keyspaceName, String typeName) {
     }
     return typeMetadata;
   }
+
+  public QueryBuilder queryBuilder() {
+    return dataStore.queryBuilder();
+  }
+
+  public Schema schema() {
+    return dataStore.schema();
+  }
+
+  public CompletableFuture<ResultSet> execute(
```

Comment:
I agree that these are sort of awkward, and I was wondering about same thing. Especially for `AuthenticatedDB`, less so for `DocumentDB`.
My main concern is that there should ideally not be multiple partially overlapping layers, unless there is some clear reason for why one layer does not encapsulate another. So the design question here is sort of like... how do we combine Authnz aspects with `DataStore`?

Having said that, if it's ok I think I'll first do this refactoring to reduce calls to `DataStore`, and then we can figure out a proper abstraction -- this because the main goal here is to separate out Docs API and REST API access.
That is needed for stargate v2 when these APIs will be separate codebases, possibly deployed separately.
I know that this does not mandate full hiding of `DataStore`, but it helps show exact data access that REST API in particular needs (which has to be re-routed to go over gRPC [or, if we preferred it, CQL protocol]).

Last thing: `QueryExecutor` is interesting... it seems that this encapsulates "extra" features Document API needs above and beyond `DataStore`. I will need to understand it better for v2 work.